public/views/pages/styleguide.liquid

```diff
@@ -1,65 +1,105 @@
 ---
 layout: modules/components/styleguide
+slug: styleguide(/:component)
 ---
 {% function components_query_result = 'modules/components/lib/queries/get_components' %}
+{% function component_search_result = 'modules/components/lib/queries/get_component', component: context.params.component %}
+{% assign component = component_search_result.results.results[0] %}
```

Comment:
What you might want to do is to have a `lib/queries/components/find_by_path` which encapsulates the logic and provides extra business logic - currently, there is no user input validation, so this endpoint could be used to extract information about any part in the system, whereas your intention was most likely to provide information about partial following certain convention

Reply:
I need something (for example a machine_name property) which can properly indentify the component and can be passed via the url.

Reply:
I modified the function to load the component by ID.